### PR TITLE
Feat(Project, StarWars):: Create the Star War App UI with API integration  

### PR DESCRIPTION
**Pull request by Harsh-Upadhyay-AT**
Summary -
Create the StarWar APP with API Integration 

Change log -
Created the StarWar APP with API Integration 

Test Case -
Home Page -

- [ ] create the card 6 cards

1. film
2. starship
3. people
4. vehicles
5. species
6. planets

- [ ] This Card section This card Shows the Item Name
- [ ] Create a separate card for each item 
- [ ] Create the View Button and this button is clickable
- [ ] If click the View button this page redirects to the Spetate module Page

in this section Create the View button. if click the button then redirect the page

**Species Page**
Create the UI
- In the header section click the species button then it redirects the page
- In the Species page fetching the API data
- Show the data from the API
- Species Page shows the
- [ ] Image
- [ ] Name
- create the view button in the cart section
- This button is clickable if click the button then open the new tab and show the species details.
- In details Page shows the list are -
- [ ] name
- [ ] classification
- [ ] designation
- [ ] average Height
- [ ] average lifespan
- [ ] eyes color
- [ ] hair color
- [ ] skin color
- [ ] language
- [ ] created
- [ ] edited
- Pagination
- [ ] show the 10 items on one page
- [ ] pre icon is disabled in-home and the next button is disabled to the last item show page
- [ ] click the number button the page redirects properly

**Films Page**

1. Click the Film Page then the data coming from API will be shown

- [ ] Image	
- [ ] Title
- [ ] Episode
- [ ] Release Date

2. Show the View button if click this button then open the details page

3. Show the Film Details page -

- [ ] Title
- [ ] Episode
- [ ] Opening Crawl
- [ ] Director
- [ ] Producer
- [ ] Release Date
- [ ] Created
- [ ] Edited
- [ ] 

4. Films details page shows the back button. This button is clickable if click the button then redirects to the films page.

5.Pagination -

- [ ]  Create the Pagination
- [ ]  Set the 10-page limit for each page
- [ ]  Create the next and previous button
- [ ]  The First Page Previous button should be disabled 
- [ ]  The Last Page Next button should be disabled
- [ ]  If click the next button then show the next

**StarShips Page -**
		
1. Click the StarShips Page then the data coming from API will be shown

- [ ] Image 
- [ ] Name 
- [ ] Model number

2.  Show the View button if click this button then open the details page

3. Show the StarShips Details page

- [ ] Name
- [ ] Model	
- [ ] Starship Class
- [ ] Manufacturer
- [ ] Cost In Credits
- [ ] Length
- [ ] Crew
- [ ] Passengers
- [ ] Max Atmosphere Speed
- [ ] MGLT
- [ ] Cargo Capacity
- [ ] Consumables
- [ ] HyperDrive Rating
- [ ] Created
- [ ] Edited

4. StarShips details page shows the back button. This button is clickable if click the button then redirects to the StarShips page.

5. Pagination -

- [ ] Create the Pagination
- [ ]  Set the 10-page limit for each page
- [ ]  Create the next and previous button
- [ ]  The First Page Previous button should be disabled 
- [ ]  The Last Page Next button should be disabled
- [ ]  If click the next button then show the next

**People Page -**
		
1. Click the People Page then the data coming from API will be shown
2. Show the People page

- [ ] image, 
- [ ] name, 
- [ ] Birth Year, 
- [ ] Gender.

3.  Show the View button if click this button then open the details page
4. Show the People Details page

- [ ] Name
- [ ] Date Of Birth
- [ ] Eye Color
- [ ] Hair Color
- [ ] Skin Color
- [ ] Gender
- [ ] Height
- [ ] Mass
- [ ] Edited
- [ ] Created

5. The people details page shows the back button. This button is clickable if click the button then redirects to the People page.

6. Pagination -

- [ ]  Create the Pagination
- [ ]  Set the 10-page limit for each page
- [ ]  Create the next and previous button
- [ ]  The First Page Previous button should be disabled 
- [ ]  The Last Page Next button should be disabled
- [ ]  If click the next button then show the next

**Vehicles** -
	1.  Click the Vehicles Page then the data coming from API will be shown
	2.  Show the Vehicles page

- [ ] image
- [ ] Name
- [ ] Model

3. Show the View button if click this button then open the details page
4. Show the Vehicles Details page

- [ ] Name
- [ ] Model
- [ ] Vehicle Class
- [ ] Consumables
- [ ] Cargo Capacity
- [ ] Cost In Credits
- [ ] Manufacturer
- [ ] Length
- [ ] Crew
- [ ] Passengers
- [ ] Max Atmosphere Speed
- [ ] Created
- [ ] Edited

5. Vehicle details page shows the back button. This button is clickable if click the     button then redirect the Vehicles page.

Pagination -

- [ ] Create the Pagination
- [ ] Set the 10-page limit for each page
- [ ] Create the next and previous button
- [ ] The First Page Previous button should be disabled 
- [ ] The Last Page Next button should be disabled
- [ ] If click the next button then show the next

**Species -**

1. Click the Species Page then the data coming from API will be shown
2. Show the Species

- [ ] image,
- [ ] Name,
- [ ] classification

3. Show the View button if click this button then open the details page

- [ ] Name
- [ ] Classification
- [ ] Designation
- [ ] Average Height
- [ ] Average Lifespan
- [ ] Eye Color
- [ ] Hair Color
- [ ] Skin Color
- [ ] Language
- [ ] Created
- [ ] Edited

4. The Species details page shows the back button. This button is clickable if the button then redirected to the Species page.

Pagination -

- [ ] Create the Pagination
- [ ] Set the 10-page limit for each page
- [ ] Create the next and previous button
- [ ] The First Page Previous button should be disabled 
- [ ] The Last Page Next button should be disabled
- [ ] If click the next button then show the next

**Planets**-

1. Click the Planets Page then the data coming from API will be shown
2. Show the Planets Details-

- [ ] Image
- [ ] Name
- [ ] Diameter
- [ ] Gravity
- [ ] Population

3. Show the View button if click this button then open the details page

- [ ] Name
- [ ] Diameter
- [ ] Gravity
- [ ] Population
- [ ] Rotation Period
- [ ] orbital Period
- [ ] Climate
- [ ] Terrain
- [ ] Surface Water
- [ ] Created
- [ ] Edited

4. Planets details page shows the back button. This button is clickable if the button is then redirected to the Planets page.

Pagination -

- [ ] Create the Pagination
- [ ] Set the 10-page limit for each page
- [ ] Create the next and previous button
- [ ] The First Page Previous button should be disabled 
- [ ] The Last Page Next button should be disabled
- [ ] If click the next button then show the next

### CODE REVIEW

File: star-war-character-app/src/pages/Species/SpeciesDetails/SpeciesDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+const SpeciesDetails = () => {
+  const { specificSpecies } = useSelector(
+    (state: IRootState) => state.speciesStateData
+  );
+  const dispatch = useAppDispatch();
+  const isLoading = useSelector(
+    (state: IRootState) => state.speciesStateData.isLoading
+  );
+  const { speciesId } = useParams();
+  useEffect(() => {
```
Comment (bhavikg-dev) [inline star-war-character-app/src/pages/Species/SpeciesDetails/SpeciesDetails.tsx:21]: Change the file name as per discussed.

File: star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { Link, useParams } from "react-router-dom";
+import { IRootState, useAppDispatch } from "../../../redux/store";
+import { useSelector } from "react-redux";
+import { Fragment, useEffect } from "react";
+import { getIndividualVehicleActions } from "../../../redux/VehicleSlice/VehicleAsyncThunk";
+import { Strings } from "../../../resource/Strings";
```
Comment (bhavikg-dev) [inline star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx:6]: Setup relative path

File: star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+              <label>{Strings.passengers}:</label> {specificVehicle.passengers}
+            </div>
+
+            <div className="detail-item">
+              <label>{Strings.maxAtmosphereSpeed}:</label>
+              {specificVehicle.max_atmosphering_speed}
+            </div>
+
+            <div className="detail-item">
+              <label>{Strings.created}:</label>{" "}
```
Comment (bhavikg-dev) [inline star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx:83]: These should be corrected, and should not be part of the code

File: star-war-character-app/src/redux/FilmSlice/FilmSlice.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    producer: "",
+    release_date: "",
+    url: "",
+    starships: [],
+    species: [],
+    planets: [],
+  }
+
+const initialState: FilmList = {
+list: [],
```
Comment (bhavikg-dev) [inline star-war-character-app/src/redux/FilmSlice/FilmSlice.tsx:23]: Code should be formatted

File: star-war-character-app/src/redux/PeopleSlice/PeopleType.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    birth_year: string;
+    gender: string;
+    homeworld: string;
+    films: string[];
+    species: string[];
+    created: string;
+    edited: string;
+  }
+  export interface PeoplesList {
+    list: Peoples[];
```
Comment (bhavikg-dev) [inline star-war-character-app/src/redux/PeopleSlice/PeopleType.tsx:19]: It should be `People`

**Review by bhavikg-dev — CHANGES_REQUESTED** [review]

File: star-war-character-app/src/pages/Species/SpeciesDetails/SpeciesDetails.tsx (hunk 1 above)
Comment (Harsh-Upadhyay-AT) [reply star-war-character-app/src/pages/Species/SpeciesDetails/SpeciesDetails.tsx:21]: I have resolved this issue in this PR https://github.com/Harsh-Upadhyay-AT/Star-War-Character-App/pull/2 Please check it.

File: star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx (hunk 2 above)
Comment (Harsh-Upadhyay-AT) [reply star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx:6]: I have resolved this issue in this PR https://github.com/Harsh-Upadhyay-AT/Star-War-Character-App/pull/2 Please check it.

File: star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx (hunk 3 above)
Comment (Harsh-Upadhyay-AT) [reply star-war-character-app/src/pages/Vehicle/VehicleDetails/VehicleDetails.tsx:83]: I have resolved this issue in this PR https://github.com/Harsh-Upadhyay-AT/Star-War-Character-App/pull/2 Please check it.

File: star-war-character-app/src/redux/FilmSlice/FilmSlice.tsx (hunk 4 above)
Comment (Harsh-Upadhyay-AT) [reply star-war-character-app/src/redux/FilmSlice/FilmSlice.tsx:23]: I have resolved this issue in this PR https://github.com/Harsh-Upadhyay-AT/Star-War-Character-App/pull/2 Please check it.

File: star-war-character-app/src/redux/PeopleSlice/PeopleType.tsx (hunk 5 above)
Comment (Harsh-Upadhyay-AT) [reply star-war-character-app/src/redux/PeopleSlice/PeopleType.tsx:19]: I have resolved this issue in this PR https://github.com/Harsh-Upadhyay-AT/Star-War-Character-App/pull/2 Please check it.

**Pull request merged**